gptqmodel/models/base.py

```diff
@@ -590,8 +591,13 @@ def forward(self, *args, **kwargs):
 
     def generate(self, **kwargs):
         """shortcut for model.generate"""
-        with torch.inference_mode(), torch.amp.autocast(device_type=self.device.type):
-            return self.model.generate(**kwargs)
+        load_format = kwargs.pop('load_format', None)
+        with torch.inference_mode():
+            if load_format == 'vllm':
+                return vllm_generate(self.model, **kwargs)
+            else:
+                torch.amp.autocast(device_type=self.device.type)
```

Comment:
`autocast` is a context? The original code is using `with autocast` but here it is just a statement.